content/blog/2020-11-30-super-optimization/index.md

```diff
@@ -0,0 +1,49 @@
+
+The paper also has proposed a probailistict execution test in which the superoptimizer picks a test vectr and see if the output results are the same between the test and generated program.  Although this test is not compprehansive, the author claims that there was never a case where probailistict execution test passes while boolean test fails.  The time it takes for exhaustive search can be futher reduced by ulitization of pruing on the generated sequences that are equivalent to shorter sequences.  
+
+## Results
+
+One of the most intresting part of this paper is the results.  Although compare to other papers where they have a long compression of performance, superoptimizer reported it's result by comparing the execution of the input porgram and assembly.  For example, given the following signum program: 
+<img src="superin.png" style="width: 30%;">
+
+Getting signum program to be immplemented in less then 9 assembly statments is already hard.  But superoptimizer managed to convert the program into a series of adds, subtracts, negation without any branches.  Resulting in the following results:
+<img src="superout.png" style="width: 30%;">
+By removing branches and the number of instructions, this immplemation has provide huge benfits to morden out of order processors (remove the need of branch predictions) and in both number of execution cycles.    Other results are similar interesting and that demestrated clearly the different then what a regular complier would generate.  
+
+## Limitations, Merits and Shortcomings
+
+One of the biggest limitaiton of super optimization would have been complier time cost and the benfits taht can be obtained at run time.  Although super optimization can ensure optimality, it does not nessciarity ensure a preformance bouns over exisitng solutions. That being said, this paper has a great impact in both llvm and gcc communities.  Despite the super optimization is not being actively used in code compliation that they each have their own super optimization immplemention, which is partly inspirted by this paper. Although this paper has strong merits on the grounds of creating optimal code in applications and usage, it is not very pratical in most contempary compliers.  
+
+
+## Connections to Modern Works
+
+Super optimizations desipte having good results has not be very wide adopted in for compliation compliers framework since this paper has been writen.  This concept has been reintroduced as papers such as [Automatic generation of peephole superoptimizers (Bansal, Aiken 2006)] [autopep]  Although the usage was wastly different from the orginal form of super optimization, this cencept managed to endure the test of time and leave its make in peepholde generation.  
```

Comment:
"compliation compliers framework" (not sure what noun phrase this is supposed to be)